src/index.ts

```diff
@@ -94,6 +109,44 @@ function getLoaderOptions(loader: interfaces.Webpack) {
 }
 
 /**
+ * Get any files generated by previous loaders in the same chain.
+ */
+function getGeneratedFiles(loader: interfaces.Webpack) {
+    const generatedFiles: { [filePath: string]: string } = {};
+    for (let i = loader.loaders.length - 1; i > loader.loaderIndex; i--) {
+        const data = loader.loaders[i].data;
+        const files = data && data['ts-loader-files'];
+        if (files) {
+            for (let filePath of Object.keys(files)) {
+                if (filePath) {
+                    generatedFiles[path.normalize(filePath)] = files[filePath];
+                }
+            }
+        }
+    }
+    return generatedFiles;
+}
+
+/**
+ * Remove dependencies specified by argument exclude.
+ */
+function removeDependencies(loader: interfaces.Webpack, exclude: (filePath: string) => boolean): void {
+    const dependencies = loader.getDependencies();
+    const contextDependencies = loader.getContextDependencies();
+    loader.clearDependencies();
+    for (const filePath of dependencies) {
```

Comment:
Could `filePath` be renamed to something so people don't mix it up with the `filePath` in the method signature for `exclude` please?